elyra/pipeline/airflow/processor_airflow.py

```diff
@@ -442,19 +440,38 @@ def create_pipeline_file(self, pipeline, pipeline_export_format, pipeline_export
                 import black
                 autopep_output = autopep8.fix_code(python_output)
                 output_to_file = black.format_str(autopep_output, mode=black.FileMode())
+
                 fh.write(output_to_file)
 
         return pipeline_export_path
 
-    def _get_unique_operation_name(self, operation_name: str, operation_list: list) -> str:
-        unique_name_counter = 1
-        unique_operation_name = operation_name
-        names = [op['notebook'] for op in operation_list]
-        while unique_operation_name in names:
-            unique_name_counter += 1
-            unique_operation_name = ''.join([operation_name, '_', str(unique_name_counter)])
+    def _create_unique_node_names(self, operation_list, parent_unique_names=None):
+        unique_names = parent_unique_names or {}
+        for operation in operation_list:
+            # if a duplicate name is found
+            if operation.name in unique_names:
+                unique_names[operation.name] += 1
+                new_name = f"{operation.name}_{unique_names[operation.name]}"
+                if new_name in unique_names:  # look-ahead
+                    self._create_unique_node_names([operation], unique_names)
+                else:
+                    operation.name = new_name
+                    unique_names[operation.name] = 1
```

Comment:
- The recursion doesn't feel right here - both in the return value behavior (previous comment), and also due to the fact that it will always be called with a single-item list.
- The "look-ahead" comment is more like "look-behind" because it only considers past names
- Perhaps a `while not unique` mechanism may be easier to follow...

```python
            if operation.name in unique_names:
                new_name = operation.name
                while new_name in unique_names:
                    new_name = f"{operation.name}_{unique_names[operation.name]}"
                    unique_names[operation.name] += 1
                operation.name = new_name
```
- I find the implicit `_1` suffix a bit counterintuitive - should the seed value be `0`? (The above example will use `_1` for the first duplicate w/o having to change the seed value from `1`.)
- I think this statement `unique_names[operation.name] = 1` could be unconditional since it must be called regardless of the route.
- Eliminating the recursion eliminates the need for the `parent_unique_names` parameter and `unique_names` could start out as an empty dict.

Reply:
Thx for the review!
- Updated to use non recursive method
- duplicate name seeding to begin at index 1
- type hinting updated and fixed for some others
- tests updated